### PR TITLE
[CI] Add code coverage report upload steps

### PR DESCRIPTION
**Pull request by vettoreldaniele**
This is a fixed version of https://github.com/GPUOpen-Drivers/llpc/pull/1600.

Add code coverage reports to pull request CI. When the CI runs configurations with code coverage (+coverage) enabled, a code coverage report is generated and uploaded to a public Google Cloud Storage bucket for inspection. The link to the code coverage report is then posted to the pull request by the CI bot.

The Google Cloud Storage bucket has been configured so that it's read-only by the public and write access is only granted to the CI bot.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

**Comment by vettoreldaniele** [comment]
> 



> Can you paste a link to a successful run on your fork? I think all you need to change is ` github.repository == 'GPUOpen-Drivers/llpc'`

I don't have access to the secrets and the bucket doesn't allow me to generate new keys - so I can't effectively test this. I can look into why I can't generate keys but it can take a while

**Comment by dstutt** [comment]
This looks great. I'd be interested to see an example too.

**Comment by vettoreldaniele** [comment]
@dstutt  Can you try approving this PR (without merging) and re-running the CI checks once these are done? Your approval should allow the workflow to pick up the secret and test the full workflow (the report should be uploaded and linked here as a comment)

**Comment by vettoreldaniele** [comment]
I was able to get this working. See https://github.com/vettoreldaniele/llpc/pull/2 for an example of the generate reports and PR comments.

**Review by kuhar — APPROVED** [review]
> I was able to get this working. See [vettoreldaniele#2](https://github.com/vettoreldaniele/llpc/pull/2) for an example of the generate reports and PR comments.

Looks amazing! Thanks for working on this, Daniele.

---

To folks with merge permissions (@amdrexu @dstutt @trenouf @JaxLinAMD): you can test this workflow by first approving it and then going to a failed run and clicking 'Re-run all checks'.

The checks that are marked as failed should turn green if everything's fine.

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
I think this needs to be merged without passing all checks here because of GitHub secret protection restrictions

**Comment by kuhar** [comment]
test this please

**Review by kuhar — APPROVED** [review]

**Review by trenouf — APPROVED** [review]
LGTM

**Comment by amdvlk-admin** [comment]
### Test summary for commit 30505f0ee39ad33a4f6026136a1cb6a2ddf9bfaa
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev aaa34e134023e703e197a48d8069cf3a6ebea629</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**
